test/TestPercentageMath.sol

```diff
@@ -165,10 +168,10 @@ contract TestPercentageMath is Test {
         uint16 percentage
     ) public {
         vm.assume(percentage <= PERCENTAGE_FACTOR);
-        // prettier-ignore
+        vm.assume(percentage == 0 || y <= (MAX_UINT256 - HALF_PERCENTAGE_FACTOR) / percentage);
         vm.assume(
-            (percentage == 0 || y <= MAX_UINT256_MINUS_HALF_PERCENTAGE / percentage) &&
-            (PERCENTAGE_FACTOR - percentage == 0 || x <= MAX_UINT256_MINUS_HALF_PERCENTAGE / (PERCENTAGE_FACTOR - percentage))
+            PERCENTAGE_FACTOR - percentage == 0 ||
+                x <= (MAX_UINT256 - y * percentage - HALF_PERCENTAGE_FACTOR) / (PERCENTAGE_FACTOR - percentage)
```

Comment:
The following is a general comment, and something I've been thinking about for the past weeks. So don't take it too litterally.

Are the test conditions really useful ? I mean if we rewrite them everytime we change the function, can we really say that it helped us maintain a certain behavior ? It is making development more difficult though, because we have to rewrite this test. I'm leaning more and more towards test that are more focused and durable, leaving behind completeness, as tests cannot be complete anyway. 

The general reasoning is that if we can write code more easily, then we can also fix the code more easily. Imagine a scenario for the same PR, but where you had to rewrite 100 tests like these (it would probably take more than a week of work in total). In this case, would you think it would be worth it to change this function at all ?

For this specific test, maybe we can restrict it to arbitrary large values like $10^{36}$

Reply:
I asked me the same question.. this make few sense, because as you said I'm just re-writing the exact same conditions than in the code itself.

Reply:
I'm not going to change everything in this PR, but let's open an issue for this, and fix it later on.

Reply:
Fair point ; I think it's a consequence of the fact that this test is a unit test whereas integration tests (those you describe) are much more generic
I agree it's taking more time to change the implementation of a function tested with unit tests (because of the issue you raised) ; though I think you're forgetting that these very tests are useful during development because they can (not always) shed light on changes and induced edge cases and forces the developer to double-check with unit testing
Most of the time, it's just double-checking
Sometimes (at least for me), it showed me some edge-cases

Reply:
These unit tests are not testing that the overall function works as expected, rather than each branch of the code are working as expected
I agree it's lacking an integration test which tests a subset of values, just as you described

Reply:
>These unit tests are not testing that the overall function works as expected, rather than each branch of the code are working as expected

yes, but we mostly copy the conditions written in the code to separate these branches, so it is easy to do something wrong

Reply:
Good points, let's discuss on the dedicated issue